controllers/remotesecret_controller.go

```diff
@@ -93,6 +110,21 @@ func (r *RemoteSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return nil
 }
 
+func (r *RemoteSecretReconciler) findRemoteSecretForUploadSecret(secret client.Object) []reconcile.Request {
+	requests := make([]reconcile.Request, 0)
```

Comment:
Can it actually happen that an upload secret for a remote secret doesn't have the name annotation?
In another words, a) I think you can leave this `nil` and let append initialize the slice but more importantly b) appending to a zero-sized slice will for sure re-allocate the slice, but if we're reasonably sure that's gonna be the case 99% of the time, why not just create the array with the capacity of 1 all the time?